packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -81,19 +102,229 @@ typedef ActionListenerCallback = void Function(Action<Intent> action);
+  /// this overridable [Action].
+  ///
+  /// The `defaultAction` argument is the [Action] to be invoked where there's
+  /// no ancestor [Action]s can't be found in `context` that handle the same
+  /// type of [Intent].
+  ///
+  /// This is useful for providing a set of default [Action]s in a leaf widget
+  /// to allow further overriding, or to allow the [Intent] to propagate to
+  /// parent widgets that also support this [Intent].
+  ///
+  /// {@tool snippet --template=stateful_widget_material}
+  /// This sample implements a custom text input field that handles the
+  /// [DeleteTextIntent] intent, as well as a US telephone number input widget
+  /// that consists of multiple text fields for area code, prefix and line
+  /// number. When the backspace key is pressed, the phone number input widget
+  /// sends the focus to the preceding text field when the currently focused
+  /// field becomes empty.
+  ///
+  /// ```dart preamble
+  /// // This implements a custom phone number input field that handles the
+  /// // [DeleteTextIntent] intent.
+  /// class DigitInput extends StatefulWidget {
+  ///   const DigitInput({
+  ///       required this.controller,
+  ///       required this.focusNode,
+  ///       this.maxLength,
+  ///       this.textInputAction = TextInputAction.next,
+  ///   });
+  ///
+  ///   final int? maxLength;
+  ///   final TextEditingController controller;
+  ///   final TextInputAction textInputAction;
+  ///   final FocusNode focusNode;
+  ///   @override
+  ///   DigitInputState createState() => DigitInputState();
+  /// }
+  ///
+  /// class DigitInputState extends State<DigitInput> {
+  ///   late final Action<DeleteTextIntent> _deleteTextAction = CallbackAction<DeleteTextIntent>(
+  ///     onInvoke: (DeleteTextIntent intent) {
+  ///       // For simplicity we delete everything in the section.
+  ///       widget.controller.clear();
+  ///     },
+  ///   );
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return Actions(
+  ///       actions: <Type, Action<Intent>>{
+  ///         // Make the default `DeleteTextIntent` handler overridable.
+  ///         DeleteTextIntent: Action<DeleteTextIntent>.overridable(defaultAction: _deleteTextAction, context: context),
+  ///       },
+  ///       child: TextField(
+  ///         controller: widget.controller,
+  ///         textInputAction: TextInputAction.next,
+  ///         keyboardType: TextInputType.phone,
+  ///         focusNode: widget.focusNode,
+  ///         decoration: InputDecoration(
+  ///           border: OutlineInputBorder(),
+  ///         ),
+  ///         inputFormatters: <TextInputFormatter>[
+  ///           FilteringTextInputFormatter.digitsOnly,
+  ///           LengthLimitingTextInputFormatter(widget.maxLength),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  ///
+  /// class _DeleteDigit extends Action<DeleteTextIntent> {
```

Comment:
Dart does not have nested classes so I had to manually pass the `State` object to the override. This isn't ideal. To implement an override that makes use of `callingAction` you can't use a `CallbackAction` either, since both `invoke` and `callingAction` are protected.